File: mcp-server/internal/domains/workflows/delete.go
package workflows

import (
	"context"
	"fmt"

	"github.com/stigmer/stigmer/mcp-server/internal/domains"
	workflowv1 "github.com/stigmer/stigmer/mcp-server/proto/ai/stigmer/agentic/workflow/v1"
	"github.com/stigmer/stigmer/mcp-server/proto/ai/stigmer/commons/apiresource"
	"github.com/stigmer/stigmer/mcp-server/proto/ai/stigmer/commons/apiresource/apiresourcekind"
	"google.golang.org/grpc"
)

// Delete removes a workflow identified by org and slug. It first resolves the
// org/slug pair to a resource ID via GetByReference, then calls the
// CommandController.Delete RPC. Both calls share a single gRPC connection.
func Delete(ctx context.Context, serverAddress, org, slug string) (string, error) {
	return domains.WithConnection(ctx, serverAddress,
		func(ctx context.Context, conn *grpc.ClientConn) (string, error) {
			desc := fmt.Sprintf("workflow %q in org %q", slug, org)

			queryClient := workflowv1.NewWorkflowQueryControllerClient(conn)
			workflow, err := queryClient.GetByReference(ctx, &apiresource.ApiResourceReference{
				Org:  org,
				Kind: apiresourcekind.ApiResourceKind_workflow,
				Slug: slug,
			})
			if err != nil {
				return "", domains.RPCError(err, desc)
			}

			cmdClient := workflowv1.NewWorkflowCommandControllerClient(conn)
			deleted, err := cmdClient.Delete(ctx, &workflowv1.WorkflowId{
				Value: workflow.GetMetadata().GetId(),
			})
			if err != nil {
				return "", domains.RPCError(err, desc)
			}

			return domains.MarshalJSON(deleted)
		})
}

File: mcp-server/src/domains/workflows/delete.ts
// Workflow delete path: resolve org/slug → id, then delete, both over a single
// shared transport.
// Go parity: mcp-server/internal/domains/workflows/delete.go.

import { createClient } from "@connectrpc/connect";
import { WorkflowSchema } from "@stigmer/protos/ai/stigmer/agentic/workflow/v1/api_pb";
import { WorkflowCommandController } from "@stigmer/protos/ai/stigmer/agentic/workflow/v1/command_pb";
import { WorkflowQueryController } from "@stigmer/protos/ai/stigmer/agentic/workflow/v1/query_pb";
import { ApiResourceKind } from "@stigmer/protos/ai/stigmer/commons/apiresource/apiresourcekind/api_resource_kind_pb";

import { withTransport } from "../client.js";
import { toProtoJson } from "../marshal.js";
import { rpcError } from "../rpcerr.js";

/** Delete a workflow by org and slug, returning the deleted workflow as protojson. */
export async function deleteWorkflow(
  serverAddress: string,
  token: string,
  org: string,
  slug: string,
): Promise<string> {
  const desc = `workflow "${slug}" in org "${org}"`;
  return withTransport(serverAddress, token, async (transport, callOptions) => {
    const query = createClient(WorkflowQueryController, transport);
    let id: string;
    try {
      const workflow = await query.getByReference(
        { org, kind: ApiResourceKind.workflow, slug },
        callOptions,
      );
      id = workflow.metadata?.id ?? "";
    } catch (err) {
      throw rpcError(err, desc);
    }

    const command = createClient(WorkflowCommandController, transport);
    try {
      const deleted = await command.delete({ value: id }, callOptions);
      return toProtoJson(WorkflowSchema, deleted);
    } catch (err) {
      throw rpcError(err, desc);
    }
  });
}
